packages/plugins/RedPacket/src/SiteAdaptor/RedPacketInjection.tsx

```diff
@@ -33,7 +34,9 @@ export function RedPacketInjection() {
     return (
         <EVMWeb3ContextProvider>
             <CompositionTypeContext.Provider value={compositionType}>
-                <RedPacketDialog open onClose={handleClose} source={source} fireflyContext={fireflyContext} />
+                <WalletGuard>
```

Comment:
We have implemented a similar feature in `<ApplicationBoard />`. Please take a look.

+ packages/shared/src/UI/modals/ApplicationBoardModal/ApplicationBoard.tsx